subprojects/platform-native/src/docs/sample-templates/native-component-with-custom-source-layout/README.adoc

```diff
@@ -0,0 +1,54 @@
+= {C Application} Custom Source Layout Configuration
+:summary: Configure custom source layout for a {C application}.
+:type: sample-chapter
+:tags: sample, {application}, sources, native, {C}, gradle
+:category: {C}
+:description: See how to customize the source layout of {an application} implemented in {C} using the Gradle Nokee plugins.
+
+NOTE: You can open this sample inside an IDE using the https://www.jetbrains.com/help/idea/gradle.html#gradle_import_project_start[IntelliJ native importer] or https://projects.eclipse.org/projects/tools.buildship[Eclipse Buildship].
+
+This sample shows how to configure the source layout of a {C application}.
+The library has no dependencies, and the build has a minimal configuration.
+
+In this sample, we are configuring source layout of {an application} implemented in {C}; however, this applies to other native components and languages as well.
+
+[source,groovy]
+----
+plugins {
+	id 'dev.nokee.c-application'
+}
+
+application {
+	cSources.setFrom('srcs')        // <1>
+	privateHeaders.setFrom('incs')  // <2>
```

Comment:
❓ This is the per-implementation languages recommendation to configure the sources. There are several ways to configure which depends on the intention. Should we make a different sample for each intention or make the sample more complex or write a guide or a mix of those solutions?

For example:
- You can cherry-pick a single file: `cSources.from('some/path/to/file.c')`
- You can cherry-pick multiple files: `cSources.from('a/file.c', 'b/file.c')`
- You can include all files under a directory: `cSources.from('folder')`
- You can glob files under a directory: `cSources.from(objects.fileTree('folder').include('**/*.c'))`
- You can overwrite vs append sources: `cSources.from(/* append */)` vs `cSources.setFrom(/* overwrite */)`
- You can add the output of a tasks: `cSources.from(tasks.create('myTask') { /* task configuration */ })`

👉 I feel we should have different samples for each example above as well as a guide that walks the user through source configuration.